dspace-api/src/main/java/org/dspace/util/SolrUpgradePre6xStatistics.java

```diff
@@ -487,8 +488,8 @@ private int updateRecords(String query) throws SolrServerException, SQLException
         sQ.addSort("type", SolrQuery.ORDER.desc);
         sQ.addSort("scopeType", SolrQuery.ORDER.desc);
         sQ.addSort("ord(owningItem)", SolrQuery.ORDER.desc);
-        sQ.addSort("id", SolrQuery.ORDER.asc);
-        sQ.addSort("scopeId", SolrQuery.ORDER.asc);
+        sQ.addSort("ord(id)", SolrQuery.ORDER.asc);
+        sQ.addSort("ord(scopeId)", SolrQuery.ORDER.asc);
```

Comment:
I'm not sure I fully understand what the `ord()` method is doing for us here.  The [Solr Guide](https://lucene.apache.org/solr/guide/6_6/function-queries.html) says that `ord`
> Returns the ordinal of the indexed field value within the indexed list of terms for that field
> Example: If there were only three values ("apple","banana","pear") for a particular field X, then ord(X) would be 1 for documents containing "apple", 2 for documents containing "banana", etc.

So, it seems like, rather than sorting based on IDs (which I understand is problematic since some are integers and some are UUIDs), we are sorting on their index in lexigraphical order?  I'm assuming this must be necessary, but I'm admitting, I'm not sure I understand the purpose here.

In any case, I'm not against this change & it seems small. I just realized that I don't completely understand it, so I had a hard time giving it an immediate +1.

Reply:
@terrywbrady : Ok, so the actual *sort order* (`asc` vs `desc`) doesn't really matter here.  All you need is to simply group requests based on the DSO.  That makes sense then.

Reply:
"Exception: Invalid shift value in prefixCoded bytes (is encoded value really an INT?)" suggests that the sort expects the field to be an integer, but something else was stored in that field of this record.  Applying ord() ensures that sort will always get an integer (the position of this value in the index, which is an integer, regardless of the type of the field value).